### ISSUE TITLE
How to get the mapping rate of minimap2

### ISSUE THREAD

**Issue by lauraht** [lh3/minimap2#482]
Hi Heng,

I have a question about the mapping rate of minimap2 and would appreciate your help.

What way would you suggest to get the mapping rate of minimap2 based on the output information from minimap2?

For example, I got the following output information from running minimap2 on a fasta input (trying to align it to a reference genome):
```
[M::worker_pipeline::1900.150*2.42] mapped 77628 sequences
[M::worker_pipeline::3165.478*2.65] mapped 77700 sequences
[M::worker_pipeline::4430.938*2.75] mapped 77175 sequences
[M::worker_pipeline::5683.285*2.80] mapped 76996 sequences
[M::worker_pipeline::7008.189*2.84] mapped 77797 sequences
[M::worker_pipeline::8220.200*2.86] mapped 77290 sequences
[M::worker_pipeline::9471.380*2.88] mapped 77470 sequences
[M::worker_pipeline::10744.079*2.89] mapped 77577 sequences
[M::worker_pipeline::10862.158*2.89] mapped 8356 sequences
```

When I added these 9 numbers together, I got 627989. But my input fasta file has 627989 reads in total. Does this mean the mapping rate of minimap2 is 100%? Are there any unmapped reads by minimap2?

I tried a set of input fasta files and all got the same result (100% mapping rate). So I suspect I might have mis-understood these output messages--- does `"mapped 77628 sequences"` really mean that 77628 reads were successfully mapped to the reference genome?

I used minimap2 version 2.2-r409, and used the `-ax splice` option to align PacBio CCS reads to a reference genome.

Thank you very much!

**Comment by lh3** [comment]
Write a script or use samtools. You should definitely update to the latest version of minimap2, BTW.

**Comment by lauraht** [comment]
Thank you so much Heng!

So you mean I should find out the number of mapped reads from the output SAM file instead of basing on those output messages, right? 

Is the following command a correct way to find the number of mapped reads (excluding those alignments marked as secondary or supplementary) from SAM output?
        `samtools view -F 0x904 -c file.sam`

By using the above command, I got 88.44% and 86.05% mapping rates for two datasets. Do these numbers look reasonable to you (for CCS reads)?

BTW, we actually used the old version of minimap2 to run many datasets about 1.5 years ago, and we now need to look into the mapping rates of these old runs.

Thank you very much for your help!
